docs/1.1-DRAFT/workflows.md

```diff
@@ -169,26 +175,70 @@ A workflow diagram may still be provided even if there is no programmatic `Softw
 * [dateCreated] the date the workflow was first created, e.g. `2020-05-23`
 * [license] identifying a [CreativeWork] that details the license for distributing or editing the workflow
 * [input] and [output] identifying contextual entities for the [FormalParameter] describing input and output parameters/variables that may be varied on different workflow executions
-* [sdPublisher] to identify the [Person] or [Organization] who has made the JSON-LD description of the workflow
-* [url] to identify a public webpage or source code repository for the workflow
+* [sdPublisher] to identify the [Person] or [Organization] who has made this JSON-LD description of the workflow
+* [url] to identify a public page or source code repository for the workflow
 * [version] to indicate the released version of this workflow file
 
-Contextual entities for [FormalParameter], referenced by `input` or `output`, SHOULD describe:
+The [ComputationalWorkflow] data entity SHOULD also describe, where relevant:
+* [citation] to identify a [CreativeWork] that should be cited for this workflow
+* [contributor] to identify [Person] or [Organization] that helped conceive the workflow
+* [creativeWorkStatus] to indicate the development state of the workflow, e.g. `"Draft"` or `"Published"`
+* [description] to further describe the workflow's purpose and method
+* [funding] to a [Grant] contextual entity that has financially supported development of the workflow
+* [hasPart] to indicate individual steps as [SoftwareApplication], [SoftwareSourceCode] or [ComputationalWorkflow] entities
+* [isBasedOn] to indicate another [ComputationalWorkflow] or [CreativeWork] this workflow is based on or derived from
+* [keywords] with comma separated list of keywords that categorise this workflow
+* [maintainer] to indicate a [Person] or [Organization] responsible for maintaining the workflow, specially if different from [creator]
+* [producer] the [Organization] or [Person] that, when different from [creator] and [maintainer], can be considered to _produce_ the workflow; typically a [Project]
+* [publisher] the [Organization] or [Person] that is making the workflow publicly available, e.g. `https://workflowhub.eu/`, `https://usegalaxy.eu/` or `https://usegalaxy.eu/`
+* [runtimePlatform] the runtime platform or script interpreter as an informal text string, e.g. `"Python2.3"`, `"Galaxy"` (see also [programmingLanguage] above)
+* [softwareRequirements] indicating [SoftwareApplication]s or [SoftwareSourceCode]s needed for this workflow, its steps and workflow engine to execute, e.g. `https://bioconda.github.io/`, `https://docs.docker.com/engine/`, `https://bio.tools/samtools`
+* [targetProduct] to indicate a [SoftwareApplication] for the operating system or product which this code applies to, e.g. `https://www.microsoft.com/en-gb/windows/get-windows-10` or `https://releases.ubuntu.com/20.04/`
```

Comment:
Yes, we could perhaps think of https://schema.org/runtimePlatform as the textual version of https://schema.org/targetProduct - it is much more important to formally link to the runtime for the workflow rather than the operating system that engine again runs on - but then many workflows will also be platform-specific.  

For instance you can run Snakemake on Windows - but a particular snakemake workflow, which runs shell commands, will probably require OSX/Linux or even a particular distribution if it is not using conda packages for dependencies. 

We are here limited to listing properties from [ComputationalWorkflow](https://bioschemas.org/profiles/ComputationalWorkflow/) profile but I added the examples -  agree we can make more workflow-specific examples.


So perhaps we need both and the workflow engine itself should be under `targetProduct` (the application that can run this workflow) while `softwareRequirements` can list what the workflow and engine needs to run it (e.g. nextflow as product would need the conda requirement for this particular workflow)

Reply:
+1. The line becomes a little fuzzy. I understand that when you are talking on code/software, it makes sense to have the relationship through targetProduct. E.g., this code is for the software "BioToolsYZ1.0", but when it comes to workflows, maybe it's just too much.